File: services/typescript/elysia/src/routes/handlers/delete.ts
import type { Context } from "elysia";
import { type State, parseId } from "app";
import { getLogger } from "telemetry";
import { AppErrors, sendError } from "../errors.js";

export function deleteHandler(state: State) {
  return ({ params, path, request, set }: Context) => {
    const logger = getLogger();

    const id = parseId(params.id);
    if (!id.success) {
      logger.warn({ method: request.method, path }, "Invalid path");
      return sendError(set, AppErrors.InvalidPath);
    }

    const task = state.tasks.get(id.data);

    if (task === undefined) {
      logger.warn({ id: id.data, method: "DELETE", path }, "Task not found");
      return sendError(set, AppErrors.TaskNotFound);
    }

    state.tasks.delete(id.data);

    logger.info(
      {
        id: id.data,
        secret: task.secret.length,
        operation: task.operation.toString().toLowerCase(),
        method: "DELETE"
      },
      "Removed task"
    );

    set.status = 204;
    return null;
  };
}

File: services/python/flask/src/routes/handlers/delete.py
import structlog
from uuid import UUID
from flask import current_app, request
from pydantic import ValidationError
from app.state import AppState
from app.params import parse_id
from app.task import Task
from routes.errors import send_error, AppErrors


def delete_handler(id: str):
    logger = structlog.get_logger()

    try:
        task_id: UUID = parse_id(id)
    except ValidationError:
        logger.warn("Invalid path", method="DELETE", path=request.path)
        return send_error(AppErrors.InvalidPath)

    state: AppState = current_app.extensions["state"]
    task: Task | None = state.tasks.pop(task_id, None)

    if task is None:
        logger.warn(
            "Task not found", id=str(task_id), method="DELETE", path=request.path
        )
        return send_error(AppErrors.TaskNotFound)

    logger.info(
        "Removed task",
        id=str(task_id),
        secret=len(task.secret),
        operation=task.operation.lower(),
        method="DELETE",
    )

    return "", 204
